config/initializers/filter_parameter_logging.rb

```diff
@@ -35,6 +35,24 @@
   filter
   startedFormVersion
 ].freeze
-Rails.application.config.filter_parameters = [lambda do |k, v|
-  v.replace('FILTERED') if v.is_a?(String) && ALLOWLIST.exclude?(k)
-end]
+
+Rails.application.config.filter_parameters = [
+  lambda do |k, v|
+    case v
+    when Hash # Recursively iterate over each key value pair in hashes
+      v.each do |nested_key, nested_value|
+        Rails.application.config.filter_parameters.first.call(nested_key, nested_value)
+      end
+    when Array # Recursively map all elements in arrays
+      v.map! { |element| Rails.application.config.filter_parameters.first.call(k, element) }
```

Comment:
Do we want to return a filtered _copy_ instead of mutating in place? Or is replacing the intention?